lib/elixir/lib/io/ansi.ex

```diff
@@ -170,6 +170,32 @@ defmodule IO.ANSI do
   @doc "Sends cursor home."
   defsequence(:home, "", "H")
 
+  @doc """
+  Sends cursor to the absolute position specified by `lines`, `columns`, 
+  with 0,0 being the top left corner.
+  """
+  def cursor(line, column) when line >= 0 and column >= 0, do: "\e[#{line};#{column}H"
+
+  @doc "Sends cursor `lines` up (defaults to 1)."
+  @spec cursor_up(integer) :: String.t()
+  def cursor_up(), do: "\e[A"
```

Comment:
I think the `@spec` will be assigned to `cursor_up/0` and the specification won't be accurate. I see two options:

```elixir
# Option 1
@spec cursor_up :: String.t()
def cursor_up, do: "\e[A"

@spec cursor_up(integer) :: String.t()
def cursor_up(lines) when lines > 0, do: "\e[#{lines}A"

# Option 2
@spec cursor_up(integer) :: String.t()
def cursor_up(lines \\ 1)
def cursor_up(1), do: "\e[A"
def cursor_up(lines) when lines > 0, do: "\e[#{lines}A"
```

I would go for "Option 2" :)

Reply:
I feel like option 1 is more faithful to the underlying implementation which might matter in the future since terminal emulators can behave in strange ways.